### PR TITLE
chore(ec2): convert tests to Jest

### PR DESCRIPTION
**Pull request by rix0rrr**
Turn the `aws-ec2` nodeunit test set into a Jest test set.

In order to speed up the conversion, I did not actually rewrite the
tests. Instead, I added a shim layer which converts between the two
APIs. That means we're free to ditch the nodeunit dependency add future
tests in Jest format, while spending minimal effort on the conversion.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ec2/test/connections.test.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import {
   Vpc,
 } from '../lib';
 
-export = {
+nodeunitShim({
```
Comment (eladb) [inline packages/@aws-cdk/aws-ec2/test/connections.test.ts:13]: Nit (don’t change anything!): I expected this to be called `jestShim` given it’s part of the nodeunit-shim module.

Comment (eladb) [reply packages/@aws-cdk/aws-ec2/test/connections.test.ts:13]: But thank god for this!

File: packages/@aws-cdk/aws-ec2/test/instance.test.ts [hunk 2]
```diff
@@ -2,10 +2,10 @@ import { expect, haveResource } from '@aws-cdk/assert';
 import { StringParameter } from '@aws-cdk/aws-ssm';
 import * as cxschema from '@aws-cdk/cloud-assembly-schema';
 import { Stack } from '@aws-cdk/core';
-import { Test } from 'nodeunit';
+import { nodeunitShim, Test } from 'nodeunit-shim';
 import { AmazonLinuxImage, BlockDeviceVolume, EbsDeviceVolumeType, Instance, InstanceClass, InstanceSize, InstanceType, Vpc } from '../lib';
 
-export = {
+nodeunitShim({
```
Comment (eladb) [inline packages/@aws-cdk/aws-ec2/test/instance.test.ts:8]: Btw why do we need to wrap each test with this? A simple `require` of the test file and producing a jest test for each exported function should have done the trick. No?

So basically you could just added a single jest test file inside the `test` directory which would find all nodeunit tests and export them as jest tests.

Comment (rix0rrr) [reply packages/@aws-cdk/aws-ec2/test/instance.test.ts:8]: Potentially.

Though we'd also need to hook the `require("nodeunit")` call.

I'd encourage you to try to do it that way when you go to convert a module :)

**Pull request merged**